src/spikeinterface/core/core_tools.py

```diff
@@ -315,48 +284,56 @@ def write_binary_recording(
+    # create a local dict per worker
+    worker_ctx = {}
+    worker_ctx["recording"] = recording
+    worker_ctx["byte_offset"] = byte_offest
+    worker_ctx["dtype"] = np.dtype(dtype)
+    worker_ctx["cast_unsigned"] = cast_unsigned
 
-    @ SAM useful for writing with time_axis=1!
-    """
-    assert file_handle is not None
-    assert recording.get_num_segments() == 1, "If file_handle is given then only deals with one segment"
+    file_dict = {segment_index: open(file_path, "r+") for segment_index, file_path in file_path_dict.items()}
+    worker_ctx["file_dict"] = file_dict
 
-    if dtype is None:
-        dtype = recording.get_dtype()
+    return worker_ctx
 
-    job_kwargs = fix_job_kwargs(job_kwargs)
-    chunk_size = ensure_chunk_size(recording, **job_kwargs)
 
-    if chunk_size is not None and time_axis == 1:
-        print("Chunking disabled due to 'time_axis' == 1")
-        chunk_size = None
+# used by write_binary_recording + ChunkRecordingExecutor
+def _write_binary_chunk(segment_index, start_frame, end_frame, worker_ctx):
+    # recover variables of the worker
+    recording = worker_ctx["recording"]
+    dtype = worker_ctx["dtype"]
+    byte_offset = worker_ctx["byte_offset"]
+    cast_unsigned = worker_ctx["cast_unsigned"]
+    file = worker_ctx["file_dict"][segment_index]
 
-    if chunk_size is None:
-        # no chunking
-        traces = recording.get_traces(segment_index=0)
-        if time_axis == 1:
-            traces = traces.T
-        if dtype is not None:
-            traces = traces.astype(dtype)
-        traces.tofile(file_handle)
-    else:
-        num_frames = recording.get_num_samples(segment_index=0)
-        chunks = divide_segment_into_chunks(num_frames, chunk_size)
+    # Open the memmap
+    # What we need is the file_path
+    num_channels = recording.get_num_channels()
+    num_frames = recording.get_num_frames(segment_index=segment_index)
+    shape = (num_frames, num_channels)
+    dtype_size_bytes = np.dtype(dtype).itemsize
+    data_size_bytes = dtype_size_bytes * num_frames * num_channels
+
+    # Offset (The offset needs to be multiple of the page size)
+    # The mmap offset is associated to be as big as possible but still a multiple of the page size
+    # The array offset takes care of the reminder
+    mmap_offset, array_offset = divmod(byte_offset, mmap.ALLOCATIONGRANULARITY)
+    mmmap_length = data_size_bytes + array_offset
+    memmap_obj = mmap.mmap(file.fileno(), length=mmmap_length, access=mmap.ACCESS_WRITE, offset=mmap_offset)
+
+    array = np.ndarray.__new__(np.ndarray, shape=shape, dtype=dtype, buffer=memmap_obj, order="C", offset=array_offset)
+    # apply function
+    traces = recording.get_traces(
+        start_frame=start_frame, end_frame=end_frame, segment_index=segment_index, cast_unsigned=cast_unsigned
+    )
+    if traces.dtype != dtype:
+        traces = traces.astype(dtype)
+    array[start_frame:end_frame, :] = traces
```

Comment:
We are loosing the time_axis logic here.
It used to be important for some format to have time_axis=1 ti be able to write the transpose version.
Even this is super uneffcient.